tests/integration/conftest.py

```diff
@@ -109,6 +115,136 @@ def factory(command, silent=False, hide_command=False):
+        """
+        self.shell(
+            "curl --create-dirs -ko /etc/rhsm/ca/redhat-uep.pem https://cdn-public.redhat.com/content/public/repofiles/redhat-uep.pem"
+        )
+        self.shell(
+            "curl -o /etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release https://www.redhat.com/security/data/fd431d51.txt"
+        )
+
+        return
+
+    def add_client_tools_repo(self):
+        """
+        Add the client tools repository to install subscription manager from.
+        """
+        version = "7-server" if SystemInformationRelease.version.major == 7 else SystemInformationRelease.version.major
+        repo_url = f"https://cdn-public.redhat.com/content/public/repofiles/client-tools-for-rhel-{version}.repo"
+
+        self.shell(f"yum-config-manager --add-repo {repo_url}")
+
+        # On CentOS 8.5 we need to replace the $releasever in the url to 8.5,
+        # otherwise the dnf will complain with dependency issues.
+        if "centos-8" in SYSTEM_RELEASE_ENV:
+            self.shell(r"sed -i 's#\$releasever#8.5#' /etc/yum.repos.d/client-tools-for-rhel-8.repo")
+
+        return
+
+    def install_package(self, package_name="subscription-manager"):
+        """
+        Install a package.
+        :param package_name: The package to be installed. Default is subscription-manager.
+        :type package_name: str
+        """
+        command = f"yum install -y {package_name}"
+        # rhn-client-tools package obsoletes subscription-manager on Oracle Linux
+        # set the obsoletes option to 0 to be able to install the package
+        if SystemInformationRelease.distribution == "oracle":
+            command += " --setopt=obsoletes=0"
+
+        return self.shell(command)
+
+    def remove_package(self, package_name="subscription-manager*"):
+        """
+        Removes a package.
+        :param package_name: The package to be installed. Default is subscription-manager*.
+        :type package_name: str
+        """
+        command = f"yum remove -y {package_name}"
+        return self.shell(command)
+
+    def remove_client_tools_repo(self):
+        """
+        Remove the client tools repository file.
+        """
+        command = "rm -f /etc/yum.repos.d/client-tools-for-rhel*.repo"
+
+        return self.shell(command)
+
+    def remove_keys_and_certificates(self):
+        """
+        Remove the SSL certificate for accessing the CDN and the redhat RPM GPG key.
+        """
+        self.shell("rm -f /etc/rhsm/ca/redhat-uep.pem")
+        self.shell("rm -f /etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release")
+
+        return
+
+    def unregister(self):
+        """
+        Remove potential leftover subscription, unregister the system.
+        """
+        # Remove potential leftover subscription
+        self.shell("subscription-manager remove --all")
+        # Remove potential leftover registration
+        self.shell("subscription-manager unregister")
+
+        return
+
+    def prepare_full_workflow(self):
```

Comment:
Maybe `setup_requirements()` sounds better? And `terdo_requirements()`? WDYT?

Reply:
Sounds better `set_up_requirements()` and `clean_up()` ? 
@kokesak

Reply:
yes!